### PR TITLE
First draft of 2.4 upgrade guide

### PR DESCRIPTION
**Pull request by markstory**
Add the most important changes coming in 2.4.0. I didn't do an exhaustive method list as I didn't feel that was as useful as giving the high level changes and rationale for those changes.

### CODE REVIEW

File: docs/en/2-4-upgrade-guide.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Upcoming removals in 3.0
+========================
+
+The following changes will arrive in 3.0, and don't have a simple deprecation
+path. Unfortunately these changes will result in hard breaks in 3.0.
+
+Carbon aliases removed
+----------------------
+
+When chronos was started Carbon had no active maintainers. We included
```
Comment (garas) [inline docs/en/2-4-upgrade-guide.rst:63]:
```suggestion
When Chronos was started Carbon had no active maintainers. We included
```

File: docs/en/2-4-upgrade-guide.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+compatiblity aliases in chronos to help users migrate from the un-maintained
+carbon library to chronos. Presently, carbon has active maintainers and we no
+longer feel the need to provide shims.
```
Comment (garas) [inline docs/en/2-4-upgrade-guide.rst]:
```suggestion
When Chronos was started Carbon had no active maintainers. We included
compatiblity aliases in Chronos to help users migrate from the unmaintained
Carbon library to Chronos. Presently, Carbon has active maintainers and we no
longer feel the need to provide shims.
```

File: docs/en/2-4-upgrade-guide.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+No longer extending DateTime
+----------------------------
+
+Historically chronos has extended PHP's ``DateTime`` classes. This has proven to
+be problematic especially for date classes. While chronos will not extend
+PHP's datetime classes or implements the ``DateTimeInterface``, if a method does
```
Comment (garas) [inline docs/en/2-4-upgrade-guide.rst]:
```suggestion
Historically Chronos has extended PHP's ``DateTime`` classes. This has proven to
be problematic especially for date classes. While Chronos will not extend
PHP's ``DateTime`` classes or implements the ``DateTimeInterface``, if a method does
```

**Comment by markstory** [comment]
Thank you for the corrections @garas

**Pull request merged**